pkg/store/bucket.go

```diff
@@ -3174,9 +3178,22 @@ func (r *bucketChunkReader) reset() {
 	for i := range r.toLoad {
 		r.toLoad[i] = r.toLoad[i][:0]
 	}
+	r.loadingChunksMtx.Lock()
+	r.loadingChunks = false
+	r.finishLoadingChks = make(chan struct{})
+	r.loadingChunksMtx.Unlock()
```

Comment:
```suggestion
	r.loadingChunksCond = sync.NewCond(&sync.Mutex{})
```

No need to initialize `r.loadingChunks` as the default value for a `bool` is `false`.